onnxruntime/core/providers/coreml/builders/impl/activation_op_builder.cc

```diff
@@ -54,6 +74,67 @@
 
 // Operator support related
 
+namespace {
+// assumes that node.OpType() == "PRelu"
+bool IsPReluOpSupported(const Node& node, const OpBuilderInputParams& input_params,
+                        const logging::Logger& logger) {
+  const auto& input_defs = node.InputDefs();
+
+  // X input rank must be at least 3
+  {
+    std::vector<int64_t> x_shape;
+    if (!GetShape(*input_defs[0], x_shape, logger)) {
+      return false;
+    }
+    if (x_shape.size() < 3) {
+      LOGS(logger, VERBOSE) << "PRelu 'X' input must have at least 3 dimensions";
+      return false;
+    }
+  }
+
+  // slope input must be constant
+  {
+    const auto& initializers = input_params.graph_viewer.GetAllInitializedTensors();
+    const auto initializer_it = initializers.find(input_defs[1]->Name());
+    if (initializer_it == initializers.end()) {
+      LOGS(logger, VERBOSE) << "PRelu 'slope' input must be an initializer tensor";
+      return false;
+    }
+  }
+
+  // slope must either:
+  // - have shape [C, 1, 1]
+  // - have 1 element
```

Comment:
Is this relative to the rank of X rather than a fixed [C, 1, 1]?

Reply:
from my reading of the spec: they specify C as the dimension at index -3. the number of elements in the slope/alpha weight can be C or 1. [C, 1, 1] is the broadcastable shape with C elements.
https://apple.github.io/coremltools/mlmodel/Format/NeuralNetwork.html#activationprelu